books/calculus/bake

```diff
@@ -0,0 +1,197 @@
+    end
+
+    page.tables('$:not(.unnumbered)').each do |table|
+      BakeNumberedTable.v1(table: table, number: "#{appendix_letter}#{table.count_in(:page)}")
+    end
+
+    page.examples.each do |example|
+      BakeExample.v1(example: example,
+                     number: "#{appendix_letter}#{example.count_in(:page)}",
+                     title_tag: 'div')
+    end
+
+    BakeAppendix.v1(page: page, number: appendix_letter)
+  end
+
+  # Ok so here we go, now let's move the solutions to the end of the book.  As you know,
+  # unlike Chemistry, Calculus has an "Answer Key" composite chapter after the appendices.
+  # Similar to how we made a chapter review composite chapter above, let's make one
+  # for the Answer Key and then iterate over the chapters, making the chapter's answer key
+  # composite page which we append to the Answer Key composite chapter.
+
+  # New direction that makes the composite chapter Answer Key, appends it at the end of the
+  # book, then returns it
+  book_answer_key = BakeBookAnswerKey.v1(book: book)
+
+  book.chapters.each do |chapter|
+    BakeChapterAnswerKey.v1(chapter: chapter,
+                            metadata_source: book.metadata,
+                            strategy: :calculus,
+                            append_to: book_answer_key)
+  end
+
+  BakeStepwise.v1(book: book)
+  BakeUnnumberedTables.v1(book: book)
+
+  book.pages.each do |page|
+    # fix table captions
+    page.search('.section-exercises').each do |section|
+      section.search('.os-problem-container > .os-table:first-child').each do |table|
+        table.add_class('first-element')
+        table.parent.add_class('has-first-element')
+      end
+
+      section.search('.os-problem-container > [data-type="media"]:first-child').each do |img|
+        img.add_class('first-element')
+        img.parent.add_class('has-first-element')
+      end
+    end
+
+    page.search('div.os-table div.os-caption-container').each do |caption|
+      caption_text = caption.first('span.os-caption [data-type="title"]')
+      next unless caption_text.present?
+
+      caption_text.add_class('os-title').name = 'span'
+      divider_aux = caption.first('span.os-divider')
+      divider_aux.append(sibling:
+        <<~HTML
+          #{caption_text&.cut}
+        HTML
+      )
+    end
+  end
+
+  BakeMathInParagraph.v1(book: book)
+  BakeIndex.v1(book: book)
+  BakeCompositePages.v1(book: book)
+  BakeFootnotes.v1(book: book)
+  BakeCompositeChapters.v1(book: book)
+  BakeToc.v1(book: book)
+
+  book.chapters.each do |chapter|
+    chapter.search('div[data-type="equation"]:not(.unnumbered)').each do |equation|
+      equation.append(child:
+        <<~HTML
+          <div class="os-equation-number">
+            <span class="os-number">(#{chapter.count_in(:book)}.#{equation.count_in(:chapter)})</span>
+          </div>
+        HTML
+      )
+    end
+
+    BakePageAbstracts.v1(chapter: chapter)
+    BakePageAbstracts.v2(chapter: chapter)
```

Comment:
hmmm.... why both?

Reply:
I need both, since v1 just prepends a title, should I call v1 from v2?